web/tests/EducationBenchmarking.Web.E2ETests/Pages/CompareYourCostsPage.cs

```diff
@@ -85,28 +86,42 @@ public void AssertImageDownload()
         {
             throw new ArgumentNullException(nameof(_download));
         }
+
         var downloadedFilePath = _download.SuggestedFilename;
         Assert.True(
             string.Equals("total-expenditure.png", downloadedFilePath, StringComparison.OrdinalIgnoreCase),
             $"Expected file name: total-expenditure.png. Actual: {downloadedFilePath}"
         );
     }
 
-    public async Task AssertDimension(string expectedValue)
+    public async Task AssertDimensionValue(string chartName, string expectedValue)
     {
-        var selectedValue =
-            await TotalExpenditureDimension.EvaluateAsync<string>(
-                "select => select.options[select.selectedIndex].text");
-        Assert.True(
-            string.Equals(expectedValue, selectedValue),
-            $"Expected dimension: {expectedValue}. Actual: {selectedValue}");
+        ILocator chart;
+        switch (chartName)
+        {
+            case "total expenditure":
+                chart = TotalExpenditureDimension;
+                break;
+            default:
+                throw new ArgumentException($"Unsupported chart name: {chartName}");
+        }
+
+        await chart.ShouldHaveSelectedOption(expectedValue);
     }
 
-    public async Task ChangeDimension(string value)
+    public async Task ChangeDimension(string chartName, string value)
```

Comment:
The switch to locate the chart is common with AssertDiminsionValue

Reply:
made the changes in the latest commit.